### PR TITLE
Readonly does not work in datepicker if edit or display format set

### PR DESCRIPTION
**Pull request by barshat7**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by dmaurya929 — APPROVED** [review]

**Review by sakshi-arora1 — APPROVED** [review]

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js [hunk 1]
```diff
@@ -73,7 +73,7 @@
 
         setModel(model) {
             super.setModel(model);
-            if (!this.#noFormats()) {
+            if (!this.#noFormats() && !model.getState().readOnly) {
```
Comment (sakshi-arora1) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js:76]: will display format work in this case?

**Review by sakshi-arora1 — CHANGES_REQUESTED** [review]
Add test case for readonly date picker with display format

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js (hunk 1 above)
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js:76]: There are multiple issuse with this,
1. readOnly can be set dynamically, so please that use-case, when readOnly would be updated at runtime updateReadOnly would be called which should enable the datePicker popup. (https://opensource.adobe.com/aem-forms-af-runtime/js-docs/classes/Field.html#readonly
2. Please use the readOnly from the model, don't use state here.

**Review by rismehta — CHANGES_REQUESTED** [review]
Check comments

**Pull request closed without merging**